### ISSUE TITLE
IDE0071: boxing in interpolated string

### ISSUE THREAD

**Issue by anreton** [dotnet/roslyn#55240]
**Version Used**:
C# 9.0
Microsoft Visual Studio Community 2019 Version 16.10.4
Roslynator.Formatting.Analyzers 1.2.1 via Nuget
[IDE0071](https://docs.microsoft.com/ru-ru/dotnet/fundamentals/code-analysis/style-rules/ide0071)
[RCS1198](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1198.md)

**Steps to Reproduce**:
```csharp
enum Foo
{
	A = 0
}

var test1 = $"Test foo = {Foo.A}."; // Get diagnostic for RCS1198. But the compiler will convert the code to this with boxing: string.Format("Test foo = {0}.", Foo.A);
var test2 = $"Test foo = {Foo.A.ToString()}."; // Get diagnostic for IDE0071. But there is no boxing.
```

**Expected Behavior**:
No diagnostic for IDE0071 in interpolated string, when using value type.

**Actual Behavior**:
Get false-positive diagnostic for IDE0071 in interpolated string, when using value type.

Discussion of this scenario with the author of the popular nuget-package Roslynator Analyzers - https://github.com/JosefPihrt/Roslynator/issues/828#issuecomment-889341173

**Comment by CyrusNajmabadi** [comment]
This is the expected behavior here.  @jnm2  for context.  

@333fred coudl we consider having the compiler not box here for the well known formattable string type, and instead preemptively call .ToString to prevent this?  It seems like a poor experience that the idiomatic code has the perf hit.

**Comment by jnm2** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/43711. See especially the comment at the end where Ben opened an issue to track improving codegen.

**Comment by 333fred** [comment]
> @333fred coudl we consider having the compiler not box here for the well known formattable string type, and instead preemptively call .ToString to prevent this? It seems like a poor experience that the idiomatic code has the perf hit.

This pattern will incur no boxing when using .NET 6.

**Comment by jnm2** [comment]
@333fred Can https://github.com/dotnet/roslyn/issues/44168 be closed then?
